sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -363,6 +363,31 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext with Be
     spark.range(10).write.orc(dir)
   }
 
+  test("column nullability and comment - write and then read") {
```

Comment:
This is from the original PR: https://github.com/apache/spark/pull/13764

In that PR, we only did the test case coverage for SQL interface. We removed the test cases for non-SQL interfaces.